nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -634,12 +639,57 @@ private void logDebugInformation() {
      */
     protected HashMap<String, String> evalDynamicProperties() throws ConfigurationException {
         HashMap<String, String> newParametersToSave = new HashMap<>();
+        evalEnvironmentVariables();
         evalLoopbackURL();
         evalServerStatusKey(newParametersToSave);
         return newParametersToSave;
     }
 
     /**
+     * Expand environment variable for properties values of the form ${env:MY_VAR}.
+     *
+     * @since 9.1
+     */
+    protected void evalEnvironmentVariables() {
+        Map<String, String> modifiedKeys = new HashMap<>();
+        Set<String> removedKeys = new HashSet<>();
+        for (Object keyObject : userConfig.keySet()) {
+            String key = (String) keyObject;
+            String value = userConfig.getProperty(key);
+
+            Matcher matcher = ENV_VALUE_PATTERN.matcher(value);
+
+            while (matcher.find()) {
+                boolean booleanValue = "?".equals(matcher.group(1));
+                String envVarName = matcher.group(2);
+                String defaultValue = matcher.group(4);
+
+                String envValue = getEnvironmentVariableValue(envVarName);
+
+                String result = "";
+                if(booleanValue) {
+                    result = StringUtils.isBlank(envValue) ? "false" : "true";
+                }else {
+                    result = StringUtils.isBlank(envValue) ? defaultValue : envValue;
+                }
+
+
+                if (StringUtils.isBlank(result)) {
+                    removedKeys.add(key);
```

Comment:
why? what is the relation with the environment variables?
There is already a mechanism to remove properties in the `saveConfiguration()` method, see https://github.com/nuxeo/nuxeo/blob/4d397896df5634615d70b2f869f8eb43a4643086/nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java#L931